File: pset6/readability/readability.py
from cs50 import get_string

# 1. prompt a user for a text
# 2. calculate the num of letters, words and sentences in the text
# (for words each space indicates a new word + 1 for the very first word)
# 3. calculate the average num of letters and sentences per 100 words (L and S)
# 4. calculate the Coleman-Liau index by formula: index = 0.0588 * L - 0.296 * S - 15.8
# 5. print the result

text = get_string("Text: ")
letters = 0
words = 1
sentences = 0


def average(x, y):
    return x * 100 / y


for char in text:
    if char.isalpha():
        letters += 1
    if char == " ":
        words += 1
    if char == "!" or char == "?" or char == ".":
        sentences += 1

l = average(letters, words)
s = average(sentences, words)

index = round((0.0588 * l) - (0.296 * s) - 15.8)

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: pset2/readability/readability.c
;l
float letters(string str), words(string str), sentences(string str);
float average(float x, float y), L, S;
void index(float l, float s);
int num_letters = 0;
int num_words = 0;
int num_sentences = 0;
bool start = false;

int main(void)
{
    //prompt a user for text input
    string text = get_string("Text: ");

    //calculate the number of letters, words and sentences
    letters(text);
    words(text);
    sentences(text);

    //calculate the average num of  letters and sentences per 100 words
    L = average(num_letters, num_words);
    S = average(num_sentences, num_words);

    //calculate the Coleman-Liau index 0.0588 * L - 0.296 * S - 15.8
    index(L, S);
}

float letters(string str)
{
    for (int i = 0, n = strlen(str); i < n; i++)
    {
        //when see an alphabetic character increase the num of letters by 1
        if (isalpha(str[i]) > 0)
        {
            num_letters ++;
        }
    }
    return num_letters;
}

float words(string str)
{
    for (int i = 0, n = strlen(str); i < n; i++)
    {
        //look for the first alphabetic character and increase the num of words by 1
        if (start == false)
        {
            if (isalpha(str[i]) > 0)
            {
                num_words ++;
                start = true;
            }
        }
        //afterwards don't run the previous code, look for spaces and increase the num of words by 1
        else
        {
            if (isspace(str[i]) > 0)
            {
                num_words ++;
            }
        }
    }
    return num_words;
}

float sentences(string str)
{
    for (int i = 0, n = strlen(str); i < n; i++)
    {
        //when see the end of sentence (indicated by !, ?, or .) increase the num of sentences by 1
        if (str[i] == '!' || str[i] == '?' || str[i] == '.')
        {
            num_sentences ++;
        }
    }
    return num_sentences;
}


float average(float x, float y)
// x = letters or sentences
// y = words
{
    return (x * 100) / y;
}

void index(float l, float s)
{
    float i = round((0.0588 * l) - (0.296 * s) - 15.8);
    //if the index is less than 1, print "Before Grade 1"
    if (i < 1)
    {
        printf("Before Grade 1\n");
    }
    //if the index is 16 or higher print Grade 16+
    else if (i >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", i);
    }
}
